packages/parameters/src/SSMProvider.ts

```diff
@@ -0,0 +1,388 @@
+          name = name.replace('/', '');
+        }
+        parameters[name] = parameter.Value;
+      }
+    }
+
+    return parameters;
+  }
+
+  protected async _getParametersByName(parameters: Record<string, SSMGetParametersByNameOptionsInterface>, throwOnError: boolean, decrypt: boolean): Promise<SSMGetParametersByNameOutputInterface> {
+    const results: SSMGetParametersByNameOutputInterface = {
+      response: {},
+      errors: [],
+    };
+
+    const sdkOptions: GetParametersCommandInput = {
+      Names: Object.keys(parameters),
+    };
+    if (decrypt) {
+      sdkOptions.WithDecryption = true;
+    }
+    
+    const result = await this.client.send(new GetParametersCommand(sdkOptions));
+    results.errors = SSMProvider.handleAnyInvalidGetParameterErrors(result, throwOnError);
+    results.response = this.transformAndCacheGetParametersResponse(result, parameters, throwOnError);
+
+    return results;
+  }
+
+  /**
+   * Slice batch and fetch parameters using GetPrameters API by max permissible batch size
+   */
+  protected async getParametersBatchByName(parameters: Record<string, SSMGetParametersByNameOptionsInterface>, throwOnError: boolean, decrypt: boolean): Promise<SSMGetParametersByNameOutputInterface> {
+    const results: SSMGetParametersByNameOutputInterface = {
+      response: {},
+      errors: [],
+    };
+
+    // Fetch each possible batch param from cache and return if entire batch is cached
+    const { cached, toFetch } = await this.getParametersByNameFromCache(parameters);
+    if (Object.keys(cached).length >= Object.keys(parameters).length) {
+      results.response = cached;
+
+      return results;
+    }
+
+    // Slice batch by max permitted GetParameters call and retrieve the ones that are not cached
+    const { response: batchResponse, errors: batchErrors } = await this.getParametersByNameInChunks(toFetch, throwOnError, decrypt);
+    results.response = { ...cached, ...batchResponse };
+    results.errors = batchErrors;
+
+    return results;
+  }
+
+  /**
+   * Fetch each parameter from batch that hasn't expired from cache
+   */
+  protected async getParametersByNameFromCache(parameters: Record<string, SSMGetParametersByNameOptionsInterface>): Promise<SSMGetParametersByNameFromCacheOutputType> {
+    const results: SSMGetParametersByNameFromCacheOutputType = {
+      cached: {},
+      toFetch: {},
+    };
+
+    for (const [ parameterName, parameterOptions ] of Object.entries(parameters)) {
+      const cacheKey = [ parameterName, parameterOptions.transform ].toString();
+      if (!this.hasKeyExpiredInCache(cacheKey)) {
+        // Since we know the key exists in the cache, we can safely use the non-null assertion operator
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        results.cached[parameterName] = this.store.get(cacheKey)!.value;
+      } else {
+        results.toFetch[parameterName] = parameterOptions;
+      }
+    }
+
+    return results;
```

Comment:
Good point!

I have changed all the functions that allowed it according to your suggestion and used more explanatory names for the others that didn't.

I have also re-formatted some of the signatures to be on multiple lines since they were very long.